.github/workflows/typespec-validation-all.yaml

```diff
@@ -0,0 +1,74 @@
+name: TypeSpec Validation All
+
+on:
+  push:
+    branches:
+      - main
+      - RPSaaSMaster
+      - typespec-next
+
+  pull_request:
+    branches:
+      - main
+      - RPSaaSMaster
+      - typespec-next
+    paths:
+      - .gitattributes
+      - .prettierrc.json
+      - package-lock.json
+      - package.json
+      - tsconfig.json
+      - eng/**
+      - specification/suppressions.yaml
+      - specification/common-types/**
+
+      # Workflow and workflow dependencies
+      - .github/workflows/typespec-validation-all.yml
+      - .github/actions/setup-node-ci/**
```

Comment:
```suggestion
```

I'm removing the PR trigger from the DevOps pipeline in https://github.com/Azure/azure-rest-api-specs/pull/31214.  If you merge first, I'll abandon that.